app/services/subscriptions/flag_refreshed_service.rb

```diff
@@ -29,10 +29,10 @@ def flag_wallets_for_refresh
         .update_all(ready_to_be_refreshed: true) # rubocop:disable Rails/SkipsModelValidations
     end
 
-    def flag_lifetime_usage_for_refresh
-      Subscription.where(id: subscription_id).includes(:lifetime_usage, plan: :usage_thresholds).find_each do
```

Comment:
I have removed the eager loading for a single model and add the `has_many through:` on subscription so instead of loading plan and usage_threshold we get a "any? query" (shown below). Also, we might not even query if one of the earlier return condition is met.

```sql
SELECT
	1 AS one
FROM
	"usage_thresholds"
	INNER JOIN "plans" ON "usage_thresholds"."plan_id" = "plans"."id"
WHERE
	"plans"."id" = $1
LIMIT
	$2;
```